services/sidecar/rollup/services/disseminator/batch_disseminator.go

```diff
@@ -190,14 +191,17 @@ func (d *BatchDisseminator) disseminateBatch(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("failed to build batch: %w", err)
 	}
-	receipt, err := d.l1TxMgr.AppendTxBatch(
-		ctx,
-		batchAttrs.Contexts(),
-		batchAttrs.TxLengths(),
-		batchAttrs.FirstL2BlockNumber(),
-		batchAttrs.TxBatchVersion(),
-		batchAttrs.TxBatch(),
-	)
+	rlpBatchData, err := rlp.EncodeToBytes(batchAttrs)
+	if err != nil {
+		return fmt.Errorf("failed to encode batch: %w", err)
+	}
+
+	// Prepend batch data with version
+	txBatchData := make([]byte, len(rlpBatchData)+1)
+	txBatchData[0] = derivation.TxBatchVersion()
+	copy(txBatchData[1:], rlpBatchData)
```

Comment:
can we use `rlp.Encode` and pass in a buffer to avoid the copy?

Reply:
fixed